### PR TITLE
feat(project description): order project descriptions by language

### PR DESCRIPTION
**Pull request by domsteinbach**
resolves DEV-2830

### CODE REVIEW

File: apps/dsp-app/src/app/project/description/description.component.ts [hunk 1]
```diff
@@ -78,6 +81,27 @@ export class DescriptionComponent implements OnInit {
...
+        if (project.description && project.description.length > 1) {
+            // sort the descriptions by language
+            project.description = this.sortDescriptionsByLanguage(project.description);
+        }
+        return project;
+    }
+
+    // returns the descriptions sorted by language
+    sortDescriptionsByLanguage(descriptions: StringLiteral[]): StringLiteral[] {
+        const languageOrder = ['en', 'de', 'fr', 'it', 'rm'];
```
[the remark this answers is not in the document]
Comment (domsteinbach) [reply apps/dsp-app/src/app/project/description/description.component.ts]: made use of the existing language list

File: apps/dsp-app/src/app/project/description/description.component.ts [hunk 2]
```diff
@@ -78,6 +81,27 @@ export class DescriptionComponent implements OnInit {
         this.loading = false;
     }
 
+    // returns the project with the descriptions sorted by language
+    projectWithSortedDescriptions(project: ReadProject): ReadProject  {
```
[the remark this answers is not in the document]
Comment (domsteinbach) [reply apps/dsp-app/src/app/project/description/description.component.ts]: indeed, done.

File: apps/dsp-app/src/app/project/description/description.component.ts [hunk 3]
```diff
@@ -78,6 +81,27 @@ export class DescriptionComponent implements OnInit {
...
+        if (project.description && project.description.length > 1) {
+            // sort the descriptions by language
+            project.description = this.sortDescriptionsByLanguage(project.description);
+        }
+        return project;
+    }
+
+    // returns the descriptions sorted by language
+    sortDescriptionsByLanguage(descriptions: StringLiteral[]): StringLiteral[] {
```
[the remark this answers is not in the document]
Comment (domsteinbach) [reply apps/dsp-app/src/app/project/description/description.component.ts]: done.

File: apps/dsp-app/src/app/project/description/description.component.ts [hunk 4]
```diff
@@ -78,6 +82,27 @@ export class DescriptionComponent implements OnInit {
         this.loading = false;
     }
 
+    // returns the project with the descriptions sorted by language
+    private projectWithSortedDescriptions(project: ReadProject): ReadProject  {
```
Comment (mdelez) [inline apps/dsp-app/src/app/project/description/description.component.ts:86]: private methods should start with an underscore

**Review by mdelez — APPROVED** [review]

**Pull request merged**